### PR TITLE
Added a forward for successfully uploaded demos

### PR DESCRIPTION
**Pull request by TheBv**
Adds a forward that gets called when a successful api request was made. 
This allows other plugins to access the url or id for the successfully uploaded demo.

### CODE REVIEW

**Comment by icewind1991** [comment]
Can you replace the regex with a simple string prefix match/substring so it doesn't add a dependency?

**Comment by TheBv** [comment]
I'll look into it but. It's just that getting the id for the log from a url that potentially has multiple slashes in it (e.g `demos.tf/demos/123`) is quite cumbersome so I wanted to avoid that.

File: include/demostf.inc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (c43721) [inline include/demostf.inc:5]: nit: extra new line :)

File: demostf.sp [hunk 2]
```diff
@@ -145,6 +157,37 @@ public ShowResponse()
...
 	return;
 }
+
+void CallDemoUploaded(bool success, const char[] demoid, const char[] url) {
+	Call_StartForward(g_hDemoUploaded);
+
+	// Push parameters one at a time
+	Call_PushCell(success);
+	Call_PushString(demoid);
+	Call_PushString(url);
```
Comment (c43721) [inline demostf.sp:192]: I assume this is modeled after logs.tf?

**Review by c43721 — CHANGES_REQUESTED** [review]

**Comment by icewind1991** [comment]
> I'll look into it but. It's just that getting the id for the log from a url that potentially has multiple slashes in it (e.g demos.tf/demos/123) is quite cumbersome so I wanted to avoid that.

The output format for success is very consistent so checking that it starts with `STV available at: ` and getting the demo id from a fixed offset should work and keep working. (the webui upload does basically the same thing)

File: demostf.sp (hunk 2 above)
Comment (TheBv) [reply demostf.sp:192]: Pretty much

**Comment by TheBv** [comment]
I went ahead and replaced the regex with string stuff.
Yes the part in regards to id's is a bit much but I wanted to make sure to cover the case were someone might end up using a different url to post too. e.g. `demos2.tf/demos/123` and this seemed like the only simple way of achieving this.

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**